src/screens/welcome/Welcome.js

```diff
@@ -94,19 +113,28 @@ class Welcome extends Component {
     this.props.resetStory();
     this.props.resetMap();
     this.props.resetPhoto();
+    // Fetch the list of file paths
     await this.stateManager.getFilePaths();
+    // If the list of file paths is empty, set this.unfinishedReportsAbsent false
     this.unfinishedReportsAbsent = this.stateManager.filePaths.length == 0;
     if (this.state.link === "") {
+      // Fetch the data from the question folder in the app file directory
       const spreadsheetExists = await RNFS.exists(
         RNFS.DocumentDirectoryPath + "/questions"
       );
+      // If the data does not exist, fetch the default spreadsheet
       if (!spreadsheetExists) {
         await this.getMobileSpreadsheet(DEFAULT_SPREADSHEET);
       }
     }
     this.setState({ loading: false });
   }
 
+  /*
+    This function checks if the state manager has a list of file paths. If there
```

Comment:
i would rephrase this to "the function checks if the state manager identified any in-progress reports"